### ISSUE TITLE
staticcheck: warn about uses of untyped integer literals as time.Duration

### ISSUE THREAD

**Issue by mvdan** [dominikh/go-tools#222]
Take this piece of code:

```
duration := someDuration
if duration == 0 {
    duration = 5
}
time.Sleep(duration)
```

This is a simplified version of something I found in a somewhat large codebase. The intent here was to use 5 seconds, i.e. `5 * time.Second`.

One can think that they'd never write such code, but no tools will warn you about it and it can go unnoticed for a while. The confusion could come from people thinking that the base unit is `time.Second`, or from them being used to `sleep 5` meaning five seconds in a POSIX shell.

If one really wants 5 nanoseconds, `5 * time.Nanosecond` is much clearer and more readable. There is absolutely no reason to use the shorter version.

These should all be flagged:

* `duration = 5`
* `durationParam(5)`
* `duration + 5` or `duration += 5` (perhaps a bit of a stretch, though)

These should not be flagged:
* `duration = 5 * time.Second`
* `duration * 5` or `duration *= 5`

Basically, any scenario where an untyped constant is used in a duration expression as an absolute value (not multiplying, dividing, etc), it's bound to be a typo.

**Comment by dominikh** [comment]
Related: SA1004.

**Comment by mvdan** [comment]
I guess you'd want to flag `time.Duration(5)` too. That could seem like a more conscious decision to use `5` as a `time.Duration`, but `5 * time.Nanosecond` is still better in every way.
